internal/expr/prepare.go

```diff
@@ -0,0 +1,73 @@
+package expr
+
+import (
+	"bytes"
+	"fmt"
+)
+
+// PreparedExpr contains an SQL expression that is ready for execution.
+type PreparedExpr struct {
+	ParsedExpr *ParsedExpr
+	SQL        string
+}
+
+type typeNameToInfo map[string]*info
+
+// prepareInput checks that the input expression corresponds to a known type.
+func prepareInput(ti typeNameToInfo, p *inputPart) error {
+	inf, ok := ti[p.source.prefix]
+	if !ok {
+		return fmt.Errorf(`unknown type: "%s"`, p.source.prefix)
```

Comment:
Instead of ˋ"%s"ˋ you can use %q. The latter will handle escaping.

That said, we should not need the quoting here as these names shoud be validated elsewhere, and perhaps we could make the debugging easier with something along the lines of:
```
cannot prepare expression: type Fooz unknown, have: Foo, Bar, Baz
```

Reply:
Changed.